man/is.sorted.Rd

```diff
@@ -0,0 +1,50 @@
+\name{is.sorted}
+\alias{is.sorted}
+\title{ Checks if input is sorted }
+\description{
+  Checks if input is sorted.
+}
+\usage{
+  is.sorted(x, by=seq_along(x), retOrd=FALSE)
+}
+\arguments{
+  \item{x}{ data.table type object or atomic vector. }
+  \item{by}{ data.table columns used to check if \code{x} is sorted by those columns. }
+  \item{retOrd}{ logical, when \code{TRUE} it will set an attribute \code{"order"} on the returned value, providing an order of \code{x}. Works only for data.table type \code{x}, not for atomic vector. }
+}
+\details{
+  Checks if the input is object is sorted. Can check also by a subset of columns provided in \code{by} argument. Can also return an order used in computation when using \code{retOrd} argument.
+}
+\note{
+  Checking sortedness is an expensive computation, and most commonly the intermediate computation, the order, could be re-used.
+  For example the following check
+
+\preformatted{
+  if (!is.sorted(DT, by="Sepal.Length"))
+    DT = DT[order(Sepal.Length)]
+}
+
+  could be written as
+
+\preformatted{
+  if (!s <- is.sorted(DT, by="Sepal.Length", retOrd=TRUE))
+    DT = DT[attr(s, "order")]
```

Comment:
This use case, wouldn't a user do ```DT[order(Sepal.Length)]``` and have nearly identical performance?

Reply:
For iris, yes.
This is basically to address @mattdowle comment in #2325 and the warning that got removed.